src/Npgsql/Internal/TypeHandlers/ArrayHandler.cs

```diff
@@ -417,7 +417,7 @@ protected override Task WriteWithLengthCustom<TAny>([DisallowNull] TAny value, N
     // us (need to handle many types of T, e.g. int[], int[,]...)
     /// <inheritdoc />
     public override Task WriteObjectWithLength(object? value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async, CancellationToken cancellationToken = default)
-        => value is null || value is DBNull
+        => value is null || value == DBNull.Value
```

Comment:
It's a mistake
value is DBNull
Right 
value == DBNull.Value

Reply:
Can you explain why?

Reply:
I was wrong